rust-runtime/aws-smithy-http/src/byte_stream.rs

```diff
@@ -628,14 +634,167 @@ mod tests {
             .await
             .expect("should have some data")
             .expect("read should not fail");
-        // The size of one read should be equal to that of the buffer size
+        // The size of one read should be equal to that of the length
         assert_eq!(some_data.len(), 16384);
 
         assert_eq!(
             ByteStream::new(body1).collect().await?.remaining(),
-            298890 - some_data.len()
+            file_length as usize - some_data.len()
         );
 
         Ok(())
     }
+
+    #[cfg(feature = "rt-tokio")]
+    #[tokio::test]
+    async fn doesnt_read_past_end_of_path_based_bytestreams_with_builder(
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        use super::ByteStream;
+        use std::io::Write;
+        use tempfile::NamedTempFile;
+
+        let mut file = NamedTempFile::new()?;
+        let line_0 = "Line 0\n";
+        let line_1 = "Line 1\n";
+
+        write!(file, "{}", line_0)?;
+        write!(file, "{}", line_1)?;
+
+        let body = ByteStream::read_from()
+            .path(&file)
```

Comment:
actually surprised this works—I would have expected that the file needed to be flushed

Reply:
I'll add a `flush` just to be certain